data_set_distances.go

```diff
@@ -0,0 +1,99 @@
+//
+// Compiled with Go version go1.9 linux/amd64 
+// Contributor : Shivendra Mishra 
+// Contact : shivendra_mishra at live dot c o m
+//
+package stats
+
+import (
+	"math"
+)
+
+// Validate data for distance calculation
+func ValidateData(dataPointX, dataPointY []float64) error {
```

Comment:
Let's make this a private function since it's only meant to be used internally. To do so just change the function name to `validateData`, Golang exposes uppercase functions, methods and variables.